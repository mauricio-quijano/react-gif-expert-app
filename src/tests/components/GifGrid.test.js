import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Tests on <GifGrid />', () => {
    const category = 'Test category';
    const gifs = [
        {
            id: 'ABC',
            title: 'The Title',
            url: 'http://127.0.0.1',
        },
        {
            id: 'DEF',
            title: 'The Other Title',
            url: 'http://127.0.0.1/other',
        },
    ];

    test('should display component correctly', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should display items when images are fetched with useFetchGif', () => {
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
});
