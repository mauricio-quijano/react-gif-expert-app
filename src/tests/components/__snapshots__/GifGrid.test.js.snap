// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifGrid /> should display component correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Test category
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Tests on <GifGrid /> should display items when images are fetched with useFetchGif 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Test category
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="The Title"
      url="http://127.0.0.1"
    />
    <GifGridItem
      id="DEF"
      key="DEF"
      title="The Other Title"
      url="http://127.0.0.1/other"
    />
  </div>
</Fragment>
`;
